arangod/RocksDBEngine/Methods/RocksDBSstFileMethods.cpp

```diff
@@ -0,0 +1,180 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Julia Puget
+////////////////////////////////////////////////////////////////////////////////
+
+#include "RocksDBSstFileMethods.h"
+
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Basics/files.h"
+#include "Basics/FileUtils.h"
+#include "Basics/RocksDBUtils.h"
+#include "Random/RandomGenerator.h"
+#include "RestServer/DatabasePathFeature.h"
+
+using namespace arangodb;
+namespace FU = basics::FileUtils;
+
+RocksDBSstFileMethods::RocksDBSstFileMethods(
+    bool isForeground, rocksdb::DB* rootDB,
+    RocksDBTransactionCollection* trxColl, RocksDBIndex& ridx,
+    rocksdb::Options const& dbOptions, std::string const& idxPath)
+    : RocksDBMethods(),
+      _isForeground(isForeground),
+      _rootDB(rootDB),
+      _trxColl(trxColl),
+      _ridx(ridx),
+      _sstFileWriter(rocksdb::EnvOptions(dbOptions), dbOptions,
+                     ridx.columnFamily()->GetComparator(), ridx.columnFamily()),
+      _cf(ridx.columnFamily()),
+      _idxPath(idxPath) {}
+
+RocksDBSstFileMethods::~RocksDBSstFileMethods() { cleanUpFiles(); }
+
+void RocksDBSstFileMethods::insertEstimators() {
+  auto ops = _trxColl->stealTrackedIndexOperations();
+  if (!ops.empty()) {
+    TRI_ASSERT(_ridx.hasSelectivityEstimate() && ops.size() == 1);
+    auto it = ops.begin();
+    TRI_ASSERT(_ridx.id() == it->first);
+
+    auto* estimator = _ridx.estimator();
+    if (estimator != nullptr) {
+      if (_isForeground) {
+        estimator->insert(it->second.inserts);
+      } else {
+        uint64_t seq = _rootDB->GetLatestSequenceNumber();
+        // since cuckoo estimator uses a map with seq as key we need to
+        estimator->bufferUpdates(seq, std::move(it->second.inserts),
+                                 std::move(it->second.removals));
+      }
+    }
+  }
+}
+
+rocksdb::Status RocksDBSstFileMethods::writeToFile() {
```

Comment:
Can we make this method return an `arangodb::Result` instead of a `rocksdb::Status`?
There is not much need for it to return a `rocksdb::Status`, because it is not an overridden method.